rbac/migration_tool/migrate.py

```diff
@@ -94,30 +96,28 @@ def migrate_role(
     return relationships, v2_role_bindings
 
 
-def migrate_users_for_groups(tenant: Tenant) -> list[common_pb2.Relationship]:
-    """Write users relationship to groups."""
-    relationships: list[common_pb2.Relationship] = []
-    for group in tenant.group_set.exclude(platform_default=True):
-        user_set: Iterable[Principal] = group.principals.all()
-        for user in user_set:
-            if (relationship := group.relationship_to_principal(user)) is not None:
-                relationships.append(relationship)
-    return relationships
+def migrate_groups_for_tenant(tenant: Tenant, replicator: RelationReplicator):
+    """Generate user relationships and system role assignments for groups in a tenant."""
+    groups = tenant.group_set.all()
+    for group in groups:
+        with transaction.atomic():
+            dual_write_handler = RelationApiDualWriteGroupHandler(
+                group, ReplicationEventType.MIGRATE_TENANT_GROUPS, replicator=replicator
+            )
+            if not group.platform_default:
+                dual_write_handler.generate_relations_to_add_principals(group.principals.all())
+            if group.system is False and group.admin_default is False:
+                system_roles = group.roles().filter(system=True)
+                if system_roles.exists() > 0:
```

Comment:
```suggestion
                if system_roles.exists():
```

Another option might be

```suggestion
                if any(True for _ in system_roles)
```

This is a little less readable but probably more efficient I think. If I understand django, it wouldn't do another query, it would just start/continuing iterating through the query, using results already cached.

Reply:
yes, I am confirming it removes one query.  Evaluation or query happens only in the condition. Thanks!